Add specs and upgrade gems

mattmueller opened the pull request:
This changeset upgrades a number of gems, moves the doorkeeper dependency to 2.0, and reworks/adds additional specs for behavior. I do not know if you're open to such large changes in dependencies but figured I'd submit anyway.

I originally forked this after reading #11 - I believe the specs should make the behavior clearer for these situations. This gem delegates the requests to an `OAuth::PasswordAccessTokenRequest` which means the following:

 * When no client credentials are passed but a valid resource owner is it will happily return an access token not belonging to any specific application.
 * When invalid client credentials are passed (regardless of resource owner) it will not create an access token.
 * When valid client credentials are passed along with a valid resource owner it will create an access token belonging to the application matching the client credentials.

This behavior seems to be appropriate for password access tokens, but there's a bit of extra discussion around that here: https://github.com/doorkeeper-gem/doorkeeper/issues/669

If you do want to prevent access tokens from being created when no client credentials are passed this would need to be built in to your `resource_owner_from_assertion` definition - just check those params are present and if not, raise an exception.

Let me know your thoughts.

tute commented:
It looks like the only change to this repo is the addition of specs, no behavior change. Also dependencies upgrades. I'm in favor of this.

Can you please squash your commits together (https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history)? Then I'll merge. Thanks!

tute commented on .rspec:
```diff
@@ -0,0 +1,2 @@
+--color
+--require spec_helper
```
I wouldn't add this configuration file.

mattmueller commented:
@tute - Thanks for considering this. I've squashed all of the commits together, removed the file you requested, and added a brief note to the README. Let me know if anything else needs to be changed!

tute commented:
Thanks!
